Tutorial: web to iOS flow through deeplinks (#11940)
PostHog/posthog.com: contents/tutorials/browser-ios-identification.mdx

When a user is not authenticated, you can't call `posthog.identify`. Instead, PostHog tracks [anonymous events](/docs/data/anonymous-vs-identified-events) using a randomly generated `distinct_id` that contains no identifiable information.

To correlate anonymous events from web to mobile, you can use deeplinks:
- Get a distinct ID from the web app using `posthog.get_distinct_id`
- Attach the distinct ID and other properties like UTM parameters to the deeplink
- When the user opens the mobile app, call `posthog.identify` or `posthog.alias` with the distinct ID and other properties

Let's go through an example with Next.js and SwiftUI. These principles apply to any other web and mobile app.

## 1. Get a distinct ID from the web app

On your Next.js app, you can get the distinct ID using `posthog.get_distinct_id`. This will work for both authenticated and anonymous users.

For Next.js you should call `posthog.get_distinct_id` in `useEffect` to ensure it's called after PostHog is initialized and right after the page is loaded.

> ```tsx focusOnLines=6-8

Author: 😄 Using the new focus lines component

---
title: How to track user flows from web to iOS
date: 2025-06-20
author:
  - vincent-ge
tags:
  - configuration
  - persons
  - product os
---

> If you use PostHog on both web and iOS apps, you can track user activity across them. When a user is authenticated on both web and mobile, their events are automatically correlated by identifying them by their account. We cover identifying users in-depth in the [identifying users documentation](/docs/product-analytics/identify).

Reviewer: ```suggestion
If you use PostHog on both web and iOS apps, you can track user activity across them. When a user is authenticated on both web and mobile, their events are automatically linked by identifying them through the same account. We cover identifying users in-depth in the [identifying users documentation](/docs/product-analytics/identify).
```

If a user is not authenticated on web or mobile, you can't identify the user by their account. 
Instead, you can track web to mobile flows using deeplinks. 

This is useful for use cases like:

- Tracking unauthenticated users across web and mobile.

- Identifying users on pre-authentication screens like onboarding and landing pages.

- Marketing campaign attribution from an unauthenticated website to an authenticated mobile app.

## Understanding cross-platform tracking

> When a user is authenticated on both web and mobile, their events are automatically correlated because they are identified by a `distinct_id` correlated to their account. This is why we recommend [calling `posthog.identify`](https://posthog.com/docs/product-analytics/identify#1-call-identify-as-soon-as-youre-able-to) on both web and mobile as soon as the user signs up or logs in.

Reviewer: ```suggestion
When a user is authenticated on both web and mobile, their events are automatically linked because they are identified by a shared `distinct_id` associated with their account. This is why we recommend [calling `posthog.identify`](https://posthog.com/docs/product-analytics/identify#1-call-identify-as-soon-as-youre-able-to) on both web and mobile as soon as the user signs up or logs in.
```

When a user is not authenticated, you can't call `posthog.identify`. Instead, PostHog tracks [anonymous events](/docs/data/anonymous-vs-identified-events) using a **randomly generated** `distinct_id` that contains no identifiable information.

> To correlate anonymous events from web to mobile, you can use deeplinks:

Author: ```suggestion
To associate anonymous events from web to mobile, you can use deeplinks:
```
